controllers/backend/volume/volume.go

```diff
@@ -55,3 +57,43 @@ func (v *volumeController) sync(key string, pvc *v1.PersistentVolumeClaim) error
 	_, err = v.volumes.Update(newVol)
 	return err
 }
+
+func (v *volumeController) Create(pvc *v1.PersistentVolumeClaim) (*v1.PersistentVolumeClaim, error) {
+	if pvc == nil {
+		return pvc, nil
+	}
+
+	if pvc.Labels["rio.cattle.io/use-templates"] == "" {
+		return pvc, nil
+	}
+
+	ns := pvc.Namespace
+	serviceName := pvc.Labels["app"]
+	templateName := strings.SplitN(pvc.Name, "-"+serviceName+"-", 2)
+	volTemplate, err := v.volumeLister.Get(ns, templateName[0])
```

Comment:
for VolumeClaimTemplates we set the following labels 
```golang
		labels := map[string]string{
			"rio.cattle.io/stack":          stack.Stack.Name,
			"rio.cattle.io/workspace":      stack.Stack.Namespace,
			"rio.cattle.io/volumetemplate": volumeTemplate.Name,
		}
```
I think from that you can find the volume template.

Reply:
I remember that the label which was set to the volumeTemplateSpec doesn't get populated into the actual PVC though...